analyses/doublet-detection/scripts/01b_detect-doublets.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+import anndata
+import scrublet as scr
+from pathlib import Path
+import pyprojroot
+
+# Function to run scrublet
+def run_scrublet(adata: anndata) -> tuple[float, str]:
+    """
+        Run scrublet on a counts matrix from an AnnData object, following https://github.com/swolock/scrublet?tab=readme-ov-file#quick-start
+        Returns a tuple of (barcodes, doublet_scores, predicted_doublets)
+    """
+
+    scrub = scr.Scrublet(adata.X)
+    doublet_scores, predicted_doublets = scrub.scrub_doublets()
+
+    # stringify
+    doublet_scores_str = [str(x) for x in doublet_scores]
+    predicted_doublets_str = [("doublet" if x else "singlet") for x in predicted_doublets]
+
+    return( (list(adata.obs_names), doublet_scores_str, predicted_doublets_str) )
```

Comment:
```suggestion
    results_df = pd.DataFrame(barcodes = adata.obs_names, scrublet_score = doublet_scores_str, srublet_prediction = predicted_doublets_str)

    return(results_df)
```
Why not just turn this into a dataframe and then you can directly save to tsv file?

Reply:
I would agree with this (but don't convert scores to strings). I thought about suggesting it, but wasn't sure whether you had `pandas` in the environment already.